agroal-narayana/src/main/java/io/agroal/narayana/RecoveryXAResource.java

```diff
@@ -53,6 +55,11 @@ public Xid[] recover(int flag) throws XAException {
     @Override
     public void commit(Xid xid, boolean onePhase) throws XAException {
         getConnectedResource().commit( xid, onePhase );
+        // Commit can be called after a recover(TMENDRSCAN) has been invoked.
+        // Close after reconnect to avoid a connection leak.
+        if ( reconnected ) {
```

Comment:
need the same for `rollback()`, no !?

Reply:
I did not find a reference on the first check, but double check is on my to-do list. Probably also add another test case which does the real whole roundtrip, but with more Narayana insights, of course.